src/test/java/uk/gov/justice/digital/config/BaseSparkTest.java

```diff
@@ -0,0 +1,193 @@
+package uk.gov.justice.digital.config;
+
+
+
+import static org.apache.spark.sql.functions.col;
+import static org.apache.spark.sql.functions.from_json;
+import static org.apache.spark.sql.functions.lit;
+import static org.apache.spark.sql.functions.schema_of_json;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.io.TempDir;
+import com.amazonaws.auth.AWSCredentials;
+import com.amazonaws.auth.profile.ProfileCredentialsProvider;
+
+public abstract class BaseSparkTest {
```

Comment:
This is a good start but can be improved. On my machine a `./gradlew clean test` takes nearly 10 minutes which is way too long.

For this PR replace before/after each with before/after all instead so we only start/stop spark *once* per test class.

After this PR take a look at having spark started and stopped once for the entire test suite. There's some useful discussion here. https://stackoverflow.com/questions/43282798/in-junit-5-how-to-run-code-before-all-tests